src/posix/platform/misc.cpp

```diff
@@ -92,13 +92,19 @@ int SocketWithCloseExec(int aDomain, int aType, int aProtocol)
     VerifyOrExit((fd = socket(aDomain, aType, aProtocol)) != -1, perror("socket(SOCK_CLOEXEC)"));
 
     VerifyOrExit((rval = fcntl(fd, F_GETFD, 0)) != -1, perror("fcntl(F_GETFD)"));
-    VerifyOrExit((rval = fcntl(fd, F_SETFD, rval | FD_CLOEXEC)) != -1, perror("fcntl(F_SETFD)"));
+    rval |= FD_CLOEXEC;
+    if (aNonBlock)
+    {
+        rval |= O_NONBLOCK;
+    }
```

Comment:
```suggestion
    rval |= aNonBlock ? O_NONBLOCK | FD_CLOEXEC : FD_CLOEXEC;
```
Why are we using `?:` operator in `#else` block but not here?